extensions/stringtemplate/deployment/src/main/java/org/apache/camel/quarkus/component/stringtemplate/deployment/StringtemplateProcessor.java

```diff
@@ -17,30 +17,23 @@
 package org.apache.camel.quarkus.component.stringtemplate.deployment;
 
 import io.quarkus.deployment.annotations.BuildStep;
-import io.quarkus.deployment.annotations.ExecutionTime;
-import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
-import io.quarkus.deployment.pkg.steps.NativeBuild;
-import org.apache.camel.quarkus.core.JvmOnlyRecorder;
-import org.jboss.logging.Logger;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import org.apache.camel.support.DefaultExchange;
 
 class StringtemplateProcessor {
 
-    private static final Logger LOG = Logger.getLogger(StringtemplateProcessor.class);
     private static final String FEATURE = "camel-stringtemplate";
 
     @BuildStep
     FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
-    /**
-     * Remove this once this extension starts supporting the native mode.
-     */
-    @BuildStep(onlyIf = NativeBuild.class)
-    @Record(value = ExecutionTime.RUNTIME_INIT)
-    void warnJvmInNative(JvmOnlyRecorder recorder) {
-        JvmOnlyRecorder.warnJvmInNative(LOG, FEATURE); // warn at build time
-        recorder.warnJvmInNative(FEATURE); // warn at runtime
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflectionWithMethods() {
+        return new ReflectiveClassBuildItem(true, false,
+                //required for values in properties
+                DefaultExchange.class.getName());
```

Comment:
Maybe we should not do this. See my comments in https://github.com/apache/camel-quarkus/issues/1807.

Reply:
If it may be security vulnerability, I'll remove corresponding test, I'll remove this build item and add into limitations, that this feature is not supported in native mode. @jamesnetherton Do you agree?

Reply:
Fine with me :+1:

Reply:
I've fixed it as I suggested. (also discovered a bug in camel component - https://issues.apache.org/jira/browse/CAMEL-15577)

Reply:
> Maybe we should not do this. See my comments in #1807.

I thought the same when I saw it, but I was not able to find the place where Claus mentioned that allowContextMapAll is a security vulnerabilty fix. Thanks for mentioning #1807 @jamesnetherton !